File: 1099. Two Sum Less Than K/main1099.cpp
#include <iostream>
#include <algorithm>
#include <vector>
#include <string>
using namespace std;

class Solution1099 {
public:
    int twoSumLessThanK(vector<int>& nums, int k) {
        sort(nums.begin(), nums.end());
        int i = 0, j = nums.size() - 1;
        int maxSum = -1;
        while (i < j) {
            int sum = nums[i] + nums[j];
            if (sum >= k) {
                j--;
            } else {
                maxSum = max(maxSum, sum);
                i++;
            }
        }
        return maxSum;
    }
};

int main() {
    vector<int> nums{34,23,1,24,75,33,54,8};
    int k = 60;
    Solution1099 sol;
    int res = sol.twoSumLessThanK(nums, k);
    cout << res << endl;
    return -1;
}

// nums = [34,23,1,24,75,33,54,8], k = 60

File: 1099. Two Sum Less Than K/main1099.py
from typing import List


class Solution1099:
    def twoSumLessThanK(self, nums: List[int], k: int) -> int:
        nums.sort()
        i, j = 0, len(nums) - 1
        res = -1
        while i < j:
            sum = nums[i] + nums[j]
            if sum >= k:
                j -= 1
            else:
                res = max(res, sum)
                i += 1
        return res

sol = Solution1099()
nums = [34,23,1,24,75,33,54,8]
k = 60
res = sol.twoSumLessThanK(nums, k)
print(res)